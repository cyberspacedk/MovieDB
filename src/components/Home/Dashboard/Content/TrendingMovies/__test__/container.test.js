import React from 'react';
import { shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import TrendingMoviesConnected from '../container';

describe('TrendingMoviesContainer ', () => {
  const store = configureStore()({
    trending: {
      loading: false,
      ids: [1],
      error: false,
    },
    database: {
      movies: { 1: { id: 1, overview: 'some text' } },
    },
  });
  store.dispatch = jest.fn();

  const wrapper = shallow(
    <BrowserRouter>
      <TrendingMoviesConnected store={store} />
    </BrowserRouter>,
  );
  const container = wrapper
    .dive()
    .dive()
    .dive()
    .dive()
    .dive()
    .dive()
    .dive();

  const instance = container.instance();

  it('Should match its snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  it('Check call lifeCycleMethod componentDidMount', () => {
    instance.componentDidMount();
    expect(store.dispatch).toHaveBeenCalledWith({ type: 'FETCH_REQUEST' });
  });

  it('Map state and dispatch to props', () => {
    expect(container.props()).toEqual(
      expect.objectContaining({
        topFilms: expect.any(Array),
        loading: expect.any(Boolean),
        error: expect.any(Boolean),
        fetchDataRequest: expect.any(Function),
      }),
    );
  });
});
