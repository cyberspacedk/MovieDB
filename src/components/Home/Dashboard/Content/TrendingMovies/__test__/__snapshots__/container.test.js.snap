// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrendingMoviesContainer  Should match its snapshot 1`] = `
<TrendingMovies
  empty={false}
  error={false}
  fetchDataRequest={[Function]}
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  loading={false}
  location={
    Object {
      "hash": "",
      "pathname": "/",
      "search": "",
      "state": undefined,
    }
  }
  match={
    Object {
      "isExact": true,
      "params": Object {},
      "path": "/",
      "url": "/",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "type": "FETCH_REQUEST",
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  topFilms={
    Array [
      Object {
        "id": 1,
        "overview": "some text",
      },
    ]
  }
/>
`;
